src/features/swap/components/SwapForm.tsx

```diff
@@ -119,46 +123,119 @@ export const SwapForm = () => {
           Swap
         </ButtonCustom>
       </Form>
+
+      <Intents intentRefs={snapshot.context.intentRefs} />
     </div>
   )
 }
 
-function renderIntentOutcome(intentOutcome: Context["outcome"]) {
-  if (!intentOutcome) {
-    return null
-  }
+function Intents({
+  intentRefs,
+}: { intentRefs: ActorRefFrom<typeof intentStatusMachine>[] }) {
+  return (
+    <div>
+      {intentRefs.map((intentRef, i, list) => {
+        return (
+          <Fragment key={intentRef.id}>
+            <Intent intentRef={intentRef} />
+            {i < list.length - 1 && <hr />}
+          </Fragment>
+        )
+      })}
+    </div>
+  )
+}
 
-  const status = intentOutcome.status
-  switch (status) {
-    case "SETTLED":
-      return null
+function Intent({
+  intentRef,
+}: { intentRef: ActorRefFrom<typeof intentStatusMachine> }) {
+  const snapshot = useSelector(intentRef, (state) => state)
 
-    case "NOT_FOUND_OR_NOT_VALID":
+  const amountIn = formatUnits(
+    snapshot.context.quote.totalAmountIn,
+    snapshot.context.tokenIn.decimals
+  )
+  const amountOut = formatUnits(
+    snapshot.context.quote.totalAmountOut,
+    snapshot.context.tokenOut.decimals
+  )
+
+  const swapInfo = `${amountIn} ${snapshot.context.tokenIn.symbol} -> ${amountOut} ${snapshot.context.tokenOut.symbol}`
+
+  const value = snapshot.value
+  switch (value) {
+    case "pending":
       return (
-        <div className="text-red-500 text-sm">
-          Missed deadline or don't have enough funds!
-          {intentOutcome.txHash == null ? null : (
-            <>
-              <br />
-              Tx: {intentOutcome.txHash}
-            </>
-          )}
-          {intentOutcome.intentHash == null ? null : (
-            <>
-              <br />
-              Intent: {intentOutcome.intentHash}
-            </>
-          )}
+        <div>
+          {swapInfo} 💤
+          <br />
+          Checking intent status... intentHash: {snapshot.context.intentHash}
         </div>
       )
-
-    case "ERR_CANNOT_OBTAIN_INTENT_STATUS":
+    case "checking":
       return (
-        <div className="text-red-500 text-sm">
-          Cannot confirm intent status! Tx: {intentOutcome.txHash} Intent:{" "}
-          {intentOutcome.intentHash}
+        <div>
+          {swapInfo} 💤
+          <br />
+          Checking intent status... intentHash: {snapshot.context.intentHash}{" "}
+          tx: {snapshot.context.txHash}
         </div>
       )
+    case "success":
+      return (
+        <div>
+          {swapInfo} ✅
```

Comment:
Why are we using emoji icons? If we’re missing any icons, we should request them from the designer.

Reply:
This is temporary, because we don't have mockups yet, because design was requested this Friday, and you were on the meeting as well :)